src/client/driver/command-executors/execute-action.js

```diff
@@ -182,14 +189,137 @@ function ensureOffsetOptions (element, options) {
+        var marginRight  = parseInt(computedStyle.marginRight, 10);
+        var marginBottom = parseInt(computedStyle.marginBottom, 10);
+
+        elementBounds.left   -= marginLeft;
+        elementBounds.top    -= marginTop;
+        elementBounds.right  += marginRight;
+        elementBounds.bottom += marginBottom;
+
+        options.marginLeft = marginLeft;
+        options.marginRight = marginRight;
+        options.marginTop = marginTop;
+        options.marginBottom = marginBottom;
+    }
+
+    elementBounds.width  = elementBounds.right - elementBounds.left;
+    elementBounds.height = elementBounds.bottom - elementBounds.top;
+
+    if (typeof options.cropLeft === 'number' && typeof options.cropRight === 'number')
+        options.cropWidth = options.cropRight - options.cropLeft;
+    else
+        options.cropWidth = options.cropWidth || elementBounds.width;
+
+    if (typeof options.cropTop === 'number' && typeof options.cropBottom === 'number')
+        options.cropHeight = options.cropBottom - options.cropTop;
+    else
+        options.cropHeight = options.cropHeight || elementBounds.height;
+
+    var { horizontalCropMode, verticalCropMode } = ElementScreenshotOptions.parseCropMode(options.cropMode);
+
+    var horizontalCropBounds = determineDimensionBounds(options.cropWidth, horizontalCropMode, options.cropX, elementBounds.width, { min: options.cropLeft, max: options.cropRight });
+    var verticalCropBounds   = determineDimensionBounds(options.cropHeight, verticalCropMode, options.cropY, elementBounds.height, { min: options.cropTop, max: options.cropBottom });
+
+    if (typeof options.cropLeft !== 'number')
+        options.cropLeft = Math.max(horizontalCropBounds.min, 0);
+
+    if (typeof options.cropRight !== 'number')
+        options.cropRight = Math.min(horizontalCropBounds.max, elementBounds.width);
+
+    if (typeof options.cropTop !== 'number')
+        options.cropTop = Math.max(verticalCropBounds.min, 0);
+
+    if (typeof options.cropBottom !== 'number')
+        options.cropBottom = Math.min(verticalCropBounds.max, elementBounds.height);
+
+    if (options.cropRight - options.cropLeft <= 0) {
+        options.cropLeft  = 0;
+        options.cropRight = elementBounds.width;
+    }
+
+    if (options.cropBottom - options.cropTop <= 0) {
+        options.cropTop    = 0;
+        options.cropBottom = elementBounds.height;
+    }
+
+    if (typeof options.scrollTargetX !== 'number')
+        options.scrollTargetX = Math.round((options.cropRight - options.cropLeft) / 2);
+
+    if (typeof options.scrollTargetY !== 'number')
+        options.scrollTargetY = Math.round((options.cropBottom - options.cropTop) / 2);
+}
 
 // Automations
 function createAutomation (elements, command) {
     var selectArgs = null;
 
-    if (command.options && 'offsetX' in command.options && 'offsetY' in command.options)
+    if (command.type === COMMAND_TYPE.takeElementScreenshot)
```

Comment:
It's better to get rid of all "takeScreenshot" stuff in this file. Let's keep it only for actions that have automations.

When you got the `takeElementScreenshot` command in the driver just split it into two parts.  Call `executeAction` for a Scroll Command and then execute any browser manipulation things.